src/flamegpu/util/detail/compute_capability.cu

```diff
@@ -47,6 +52,56 @@ bool compute_capability::checkComputeCapability(int deviceIndex) {
     }
 }
 
+std::vector<int> compute_capability::getNVRTCSupportedComputeCapabilties() {
+// NVRTC included with CUDA 11.2+ includes methods to query the supported architectures and CUDA from 11.2+
+// Also changes the soname rules such that nvrtc.11.2.so is vald for all nvrtc >= 11.2, and libnvrtc.12.so for CUDA 12.x etc, so this is different at runtime not compile time for future versions, so use the methods
+#if (__CUDACC_VER_MAJOR__ > 11) || ((__CUDACC_VER_MAJOR__ == 11) && __CUDACC_VER_MINOR__ >= 2)
+    nvrtcResult nvrtcStatus = NVRTC_SUCCESS;
+    int nvrtcNumSupportedArchs = 0;
+    // Query the number of architecture flags supported by this nvrtc, to allocate enough memory
+    nvrtcStatus = nvrtcGetNumSupportedArchs(&nvrtcNumSupportedArchs);
+    if (nvrtcStatus == NVRTC_SUCCESS && nvrtcNumSupportedArchs > 0) {
+        // prepare a large enough std::vector for the results
+        std::vector<int> nvrtcSupportedArchs = std::vector<int>(nvrtcNumSupportedArchs);
+        assert(nvrtcSupportedArchs.size() >= nvrtcNumSupportedArchs);
+        nvrtcStatus = nvrtcGetSupportedArchs(nvrtcSupportedArchs.data());
+        if (nvrtcStatus == NVRTC_SUCCESS) {
+            // Return the populated std::vector, this should be RVO'd
+            return nvrtcSupportedArchs;
+        }
+    }
+    // If any of the above functions failed, we have no idea what arch's are supported, so assume none are?
+    return {};
+// Older CUDA's do not support this, but this is simple to hard-code for CUDA 11.0/11.1  (and our deprected CUDA 10.x).
+// CUDA 11.1 suports 35 to 86
+#elif (__CUDACC_VER_MAJOR__ == 11) && __CUDACC_VER_MINOR__ == 1
+    return {35, 37, 50, 52, 53, 60, 61, 62, 70, 72, 75, 80, 86};
```

Comment:
These are kinda grim, but not sure of a nicer option (besides just returning 35 by default).

Reply:
I aggree it is grim, but there is no way to programatically access request these values from nvrtc at runtime for these CUDA versions. 
Providing the full vector is the better choice, so that more recent features are used where appropriate.